gci/src/main/java/com/gcinterceptor/gci/GarbageCollector.java

```diff
@@ -0,0 +1,11 @@
+package com.gcinterceptor.gci;
+
+/**
+ * Abstract garbage the runtime's garbage collector. Meant to be used by tests.
+ *
+ * @author danielfireman
+ */
+@FunctionalInterface
+public interface GarbageCollector {
```

Comment:
No need for a file for this interface. It is totally fine for this to be part of the HeapMonitor. Also, please make it private.